contracts/Nitro.sol

```diff
@@ -8,6 +8,134 @@ contract Nitro {
         return a > b ? b : a;
     }
 
+    /**
+     * @dev Computes the new outcome that should be stored against a target channel after a claim is made on its guarantor.
+     * @param initialGuaranteeOutcome the outcome containing at least one guarantee(s) which will be claimed for each asset.
+     * @param initialHoldings initial quantity of each asset held on chain for the guarantor channel. Order matches that of initialGuaranteeOutcome.
+     * @param targetChannel the index of the guarantee in the list of guarantees for the given asset -- equivalent to declaring a target channel
+     * @param initialTargetOutcome initial outcome stored on chain for the target channel.
+     * @param exitRequest list  of indices expressing which destinations in the allocation should be paid out for each asset.
+
+     */
+    function claim(
+        ExitFormat.SingleAssetExit[] memory initialGuaranteeOutcome,
+        uint256[] memory initialHoldings,
+        uint48 targetChannel,
+        ExitFormat.SingleAssetExit[] memory initialTargetOutcome,
+        uint48[][] memory exitRequest
+    )
+        public
+        pure
+        returns (
+            ExitFormat.SingleAssetExit[] memory updatedTargetOutcome,
+            uint256[] memory updatedHoldings,
+            ExitFormat.SingleAssetExit[] memory exit
+        )
+    {
+        require(initialTargetOutcome.length == initialHoldings.length);
+
+        exit = new ExitFormat.SingleAssetExit[](initialTargetOutcome.length);
+        // For asset of initialGuaranteeOutcome
+        for (
+            uint256 assetIndex = 0;
+            assetIndex < initialGuaranteeOutcome.length;
+            assetIndex++
+        ) {
+            //TODO: require is guaratn
+            ExitFormat.Allocation[] memory guarantees =
+                initialGuaranteeOutcome[assetIndex].allocations;
+
+            ExitFormat.Allocation[] memory targetAllocations =
+                initialTargetOutcome[assetIndex].allocations;
+
+            // If exitRequest is empty for teh allocation we want ALL to exit
+            ExitFormat.Allocation[] memory exitAllocations =
+                new ExitFormat.Allocation[](
```

Comment:
I think we need to update `transfer` to do something similar so it handles the case of `exitRequest=[[]]`